### ISSUE TITLE
Add categories

### ISSUE THREAD

**Issue by talha131** [Pelican-Elegant/elegant#314]
Currently we have all articles filed under "Elegant - Pelican Theme". Originally documentation was hosted at my personal blog. It made sense to file everything under "Elegant" category.

But this needs to be expanded now. Off the top of my head

1. Configuration 
1. Guides (It will have articles on integrating plugins with the theme)
1. Customization
1. Analytics

Suggestions and PR are welcomed.

**Comment by iranzo** [comment]
@talha131 as it is now under documentation/content and then subfolders I think this can be closed

**Comment by talha131** [comment]
Right. I did add some folders in docs repo.

**Comment by silverhook** [comment]
The articles are still in wrong folders. Reopening.

**Comment by talha131** [comment]
This is our next branch 

https://github.com/Pelican-Elegant/elegant/tree/next/documentation/content

New categories have been added, articles moved and extracted. Please point out specifically which articles are in wrong folder.

**Comment by silverhook** [comment]
Will provide a PR in the next days, I hope.

**Comment by talha131** [comment]
@silverhook I have moved around the articles. Kindly checkout the latest release. And let me know if I missed something. If not then please close this issue.

**Comment by silverhook** [comment]
Has anyone thought about how this would look if we moved a lot of this into pages (instead of blog posts)?

**Comment by talha131** [comment]
@jackdewinter feel free to share you feedback.

@jackdewinter had shared some ideas about categories with me via email. This gist was that 

1. We have too many categories
1. We should have clear categories so that new user would know in which section he would find the relevant help article

----
> Has anyone thought about how this would look if we moved a lot of this into pages (instead of blog posts)?

@silverhook well @jackdewinter has been working on a 101 help article. The idea is to have one or few more articles for absolute beginners. These 101 article(s) will point user to the detailed help articles.

Here is one such article

https://elegant.oncrashreboot.com/unique-home-page-features

There was a suggestion of creating a new 101 category. But perhaps we can instead create a page for 101 article.

Lets see what @jackdewinter has to say about it.

**Comment by silverhook** [comment]
> We have too many categories
> We should have clear categories so that new user would know in which section he would find the relevant help article

I agree on both fronts. Looking forward to @jackdewinter’s feedback here.

**Comment by talha131** [comment]
Continuing discussion from https://github.com/Pelican-Elegant/elegant/pull/452

>> Moving files into folders is not enough for changing the category. You will have to change the category too

> It should be enough to have just one or the other, having both seems to cause needless confusion and duplication. In that case I would suggest we simply strip the category field from the posts and categorise by folders, as this would be much easier to manage when the articles start piling up.

1. Folders do not work as categories because we have set `USE_FOLDER_AS_CATEGORY` to false
1. Folders are not suitable to work as categories. Folders depend on file system. You know well file system does not allow certain special characters to be used as name. Therefore, we will never be able to use those characters in our categories
1. Using Folders means if we have a category in Chinese or RTL like Arabic, it will not appear nicely inside the terminal. (I have experienced it first hand with Urdu and Arabic folder and file names)
1. Using folders means, to change a category we have to move file. Git does not work nicely with move and renames even if `--follow` option, so I am warry of renaming and move files

> I’m confused with the Development category. The articles there all seemed to be about how to deploy (in a CI/CD manner) Pelican and Elegant, not about how to contribute to the project (which has its own category) or develop it further locally. In the end we could also just have “Miscellaneous” as a catch-all for whatever we can’t categorise otherwise.

I agree. Currently we have only two articles in Development category. @iranzo wrote them. At that time we used them for our CI. We can either move them to 

1. Miscellaneous
1. Or to Contributing category and specify that these tips are no longer used.

**Comment by talha131** [comment]
Lets take one thing at a time.

## Item 1

**Which category best suits for the two articles present in Development category?**

My suggestion

1. Move them to Contributing
1. Delete Development folder
1. Add note inside those two articles that Elegant CI does not  used them anymore.

## Item 2

**Should we remove all the folder inside content folder?**

I have been thinking of removing all the folders from content folder.

**Why?**

1. Folders confuse contributors to think that the serve as categories which isn't true
1. Internal linking will become easier
1. If you rename a folder then you have to search for its usage in all the Markdown files and rename it too. If we remove folders then we will not have worry about it

**Why not?**

1. If you have lots of files in a folder, then Github website does slows down and in some cases refuses to show the contents

I am in between. 

## Item 3

Lets take one category at a time and fix it.

Starting with **Monetization**.

Is everyone okay with it?

1. Does it need to be rename?
1. Do we need to move article out from it?
1. Do we need to add an article to it?

**Comment by iranzo** [comment]
Rename files to contain category-topic but remove folders or just all in
same folder.

Easier to grep, sed, etc

**Comment by talha131** [comment]
> Rename files to contain category-topic but remove folders or just all in
same folder.

If we rename files to contain category topic then we will again have to rename files whenever we change the category.

I lke "just all in the same folder", regardless of category idea.

**Comment by jackdewinter** [comment]
- Item 1: Am I missing something? The only thing in that directory that I can see is the new status doc?
- Item 2: I like putting things in one directory, but concerned about all 57 articles being in the one directory.
- Item 3: Instead of Monetization, I would like to see a category named "Advanced Concepts?" with Monetization, Mailing List, Analytics and Deployment.  From my point of view, once you have your web site set up, these are things you want to do to make it go further.   Also, add an article on setting up a sitemap and robots.txt to make the SEO work better.

Thoughts?

**Comment by talha131** [comment]
> Item 1: Am I missing something? The only thing in that directory that I can see is the new status doc?

Probably due to merge conflicts, those articles have moved to "Deployment and CD folder", though their categories have not changed.

> Item 2: I like putting things in one directory, but concerned about all 57 articles being in the one directory.

So you agree. @iranzo has already liked the idea. Now lets see what @silverhook has to say.

> Instead of Monetization, I would like to see a category named "Advanced Concepts?" with Monetization, Mailing List, Analytics and Deployment. From my point of view, once you have your web site set up, these are things you want to do to make it go further.

I disagree. Look, if we have 2, 3 or more articles that have a common theme, we should move them together. All articles under monetization (plus a possibility of adding AdSense) are clearly on the same topic.

I do think, we can change "Monetization" to something easier like "Earn Money From Your Site". Monetization sounds kinda technical.

Goal is not to reduce categories for the sake of reduction or to add them just cause we like new categories. Goal is to help user find them. Montization is such a clear category that I don't think it should be merged.

> Also, add an article on setting up a sitemap and robots.txt to make the SEO work better.

Agree. We need better instructions for it.

**Comment by talha131** [comment]
I toyed with the idea of moving all the files out of folders into the root content folder. It does not work out well. We have over 80 articles. 

Best course of actions seems to settle on categories soon.

**Comment by jackdewinter** [comment]
Here are my notes from taking a look at Categories.  My big takeaway: In my opinion, we should define better Categories and Tags.  This means clearly identifying what is a category and what is a tag.

From there, it is a question of organizing it so that someone can find things. So I asked myself some basic questions about various things in the documentation:
- I am looking for the **<blank>** feature.  Where is it?
- What else can I add to writing an article?
- Are there any website features I am not taking advantage of?
- Why did <x> decide to do it this way?

**NOTE: Rough proposal: just a starting point**
- tried to come up with an organization that would work (from my point of view) for any user

- Categories and Tags seem to be doing double duty
  - Categories should be used for overarching concepts:
    - Design
	- Setup & Tuning of Websites
	- Using Elegant to Write Better Articles
	- Reaching Your Audience
  - Tags for specifics
    - you go to the Google Analytics page, see a tag for "analytics3", so what are the other 2 options
  - i.e. Google Analytics; Category=Websites; Tags=google analytics,analytics
  - i.e. FavIcon Support; Category=Websites; Tags=favicon, website icons
  - i.e. Reading Time Support; Category=Articles; Tags=reading time, Flesch-kincaid, reading ease
- not including Contributing and Development
  - smaller % of users of Elegant, done later
  - suggest adding a "how to write a new feature" article
  - links to relevant articles, such as assets-plugin.md
  
- reworking landing page
  - make sure clean departures to the various sections and their landing pages
  - alternate to navigation by categories
    - for someone looking at Elegant, we want to tell and control the story
- Elegant Design
  - why-look-and-feel.md. Can make this list first for this category?
  - focus at a high level on what elements are in Elegant, why they are done that way
  - should have no settings in here, redirect to other articles
  - move:
     - Taxonomy
     - retool A&S articles to describe why these adhere to Elegant
       - page-title.md
	   - warning-admon.md -> describe why you want this
	   - web-safe-fonts.md
  - testimonials.md -> not sure on this. want these up and front
- Elegant Websites
  - move Analytics in here, possibly with teaser page
  - move Landing page in here, possibly with teaser page
  - move Footer in here, possibly with teaser page
  - modified-date.md
  - article-redirect.md
  - author-blurbs.md
  - favicons.md
  - custom-404.md
  - move most Supported Plugins in here, possibly with teaser page
  - move Search in here
  - customize-elegant.md -> goes against development, may want to address that
- Writing Elegantly
  - write a page with a visual for the parts of the Elegant page
    - for each section of the page, list what relevant pages contain the info in nice form with links
	  - i.e. left sidebar: TOC, right sidebar:reading-time,series,etcs
  - metadata.md -> make sure current, links to other articles

  - reading-time.md
  - article-subtitle.md
  - table-of-contents.md
  - warning-admon.md -> focus more on the how, refer to design
  - move Code Snippets in here, possibly with teaser page
    - bring up to date
- Elegantly Reaching Your Audience
  - landing page to describe it better
  - SEO and SMO
  - share-post.md from Supported Plugins
  - move Comments in here
- Mailing and Monetizing (need better name)
  - possibly pull into previous
  - Monetization and Mailing List categories
- advanced? or merge with development?
  - change-labels
  - custom-style.md

**Comment by talha131** [comment]
Jack, 

I read your comment on Github. Before we proceed, let me share my ideas with you.

We will end up with 7 categories. (plus 1 or two more if I have overlooked something.)

## Analytics, SEO and SMO

We will merge Analytics and "SEO & SMO" into it.

One less category.

## Components

This category will contain all the articles about components we use like

1. Table of Conents
1. Subtitle
1. Tables (to be written)
1. Warnings and Admonition
1. Code snippets
1. Footer
1. Landing Page

3 categories are removed with this

## Connect with your readers

This needs a better title.

1. Comments
1. News Letter
1. Social Links

It will reduce one category.

## Earn money from your blog

This needs a better title.

This means renaming Monetization to "Earn money from your blog".

## Advanced

1. Search
1. 404
1. Extra customizations
1. Fav icons
1. Article redirection

## Contributing

1. Merge Design Philosophy into it.
1. Merge development into it

## Release Notes

1. Keep it

**Comment by jackdewinter** [comment]
Existing pages that I don't see referenced in your list:
- modified-date.md
- page-title.md
- web-safe-fonts.md
- author-blurbs.md
- change-labels.md
- meta-data.md  <-- I love this as a cheat sheet when publishing an article
- reading-time.md
- Supported Plugins category
- Taxonomy category

Category Names:
- "Connect with your readers" -> "Connecting With Your Readers"
  - I think this is straight to the point
- "Earn money from your blog" -> "Monetizing Your Blog"
  - a quick google search for me revealed more "monetizing" in the results than "money"

Categories:
- I am slight concerned that merging Design Philosophy in to Contributing will hide it from people that are looking for an answer to "why did they do it that way?"

**Comment by talha131** [comment]
@jackdewinter thank you for being so thorough. 

> modified-date.md

advanced

> page-title.md

components

> web-safe-fonts.md

components

> author-blurbs.md

suported plugins

> change-labels.md

advanced

> meta-data.md

advanced

> reading-time.md

Supported Plugins 

> Taxonomy category

Components

> Category Names:

Good suggestions. Agree.

> I am slight concerned that merging Design Philosophy in to Contributing will hide it from people that are looking for an answer to "why did they do it that way?"

I agree but that's the only category it can fit. I don't think we are going to add more articles to "Design Philosophy".

Moreover, if a contributors feels like changing the style, he is more likely to look it up under "Contributing" category.

So in conclusion, "Contributing" would do fine.

**Comment by talha131** [comment]
I will make these changes.

I also need to add a test for broken links so that if links are dead due to migration, we would know.

**Comment by jackdewinter** [comment]
It looks like we have supported plugins again.  Am I reading that correctly?  If so, and it's the Plugin category or another category, which one wins?  An example of that is TOC, where we need the extract_toc plugin to be able to move it.  

Other than that... "Advanced Features" instead of "advanced"?

**Comment by talha131** [comment]
> It looks like we have supported plugins again. Am I reading that correctly?

Yes. We have.

> An example of that is TOC, where we need the extract_toc plugin to be able to move it.

It should go into "Components" category. Because user cannot use `extract` plugin unless he is interested in TOC.

Similarly user needs "tipue" plugin for search to work. It will stay in "Advanced feature" b/c user cannot possibly need tipue plugin without search feature. 

So in these cases, user is expected to look for feature, rather than the plugin.

If you have doubts, then we can use subtitle, i.e. mention plugin in the subtitle of the article.

> Other than that... "Advanced Features" instead of "advanced"?

Agree.
